### PR TITLE
T007/22: Switch to fingerpring generator

### PR DESCRIPTION
**Pull request by hamzaibrahim21**
## Description
Switching fingerprint generators in all talktorials according to [#330](https://github.com/volkamerlab/teachopencadd/pull/330).

## TODO

- [x] T004 --- updated by Greg in https://github.com/volkamerlab/teachopencadd/pull/330
- [x] [T007_compound_activity_machine_learning](https://github.com/volkamerlab/teachopencadd/tree/master/teachopencadd/talktorials/T007_compound_activity_machine_learning)
- [x] [T022_ligand_based_screening_neural_network](https://github.com/volkamerlab/teachopencadd/tree/master/teachopencadd/talktorials/T022_ligand_based_screening_neural_network)
- [x] Checked if other talktorials 1-28 need switching to fingerprint generators -> no
- [x] Checked if DL-edition talktorials  need switching to fingerprint generators -> no

## Status
- [x] Ready to go

### CODE REVIEW

**Comment by dominiquesydow** [comment]
Hi @hamzaibrahim21, 

Thanks a lot for taking on this task 🎉  
I have two quick questions before looking at this PR in more detail:
- the PR description covers more talktorials than are actually changed (I can see changes for T004/7/22), what is meant by the description, or are some talktorials still missing?
- Greg already updated T004 [here](https://github.com/volkamerlab/teachopencadd/pull/330) and from this description, it sounds like he also made more changes to T004, do we have two versions of T004 now or did you repeat Greg's work in this PR?

I am adding @AndreaVolkamer for visibility.

**Comment by dominiquesydow** [comment]
Hi @hamzaibrahim21,

Thanks a lot for the additional context! :)

> All talktorials are checked and the DL-edition as well, so I made an extra box for them, although they have no changes.

I updated the PR description accordingly, thanks!

> I repeated Greg's work, as it'd be easier to push all changes to one branch, that is branched from dev branch (If I'm not mistaken).

Repeating someone else's work is I think not ideal, as they have put in work to do this (and should get the credit on GitHub in terms of commits) and it costs you time as well. Did you redo it 1-to-1? If so, we could also merge Greg's branch into this PR here, just to make sure his contribution will be visible on GitHub.

**Comment by hamzaibrahim21** [comment]
I thought that would easier for controling/merging branches later and my branch would be merged easily without problems with his branch, sorry for such misunderstanding!!
I can reset changed talktorial 04 if needed.

**Comment by dominiquesydow** [comment]
Hi @hamzaibrahim21, no worries at all! My apologies for adding complexity to this PR but I do prefer resetting T004 (could you do that?) and merging Greg's PR into this one (I am going to do that). I think you are right that we might run into conflict issues but I will give solving them a try so that we can include Greg's commit history included. Fingers crossed :)

**Comment by hamzaibrahim21** [comment]
@dominiquesydow Great! now T04 should be reset, if anything I should do or take care of, please let me know.

File: teachopencadd/talktorials/T007_compound_activity_machine_learning/talktorial.ipynb [hunk 1]
```diff
@@ -210,7 +210,7 @@
   },
```
Comment (dominiquesydow) [inline teachopencadd/talktorials/T007_compound_activity_machine_learning/talktorial.ipynb:210]: <pre class="ql-syntax" spellcheck="false">Line #15.    from rdkit.Chem import MACCSkeys , rdFingerprintGenerator
</pre><p>as</p><pre class="ql-syntax" spellcheck="false">Line #15.    from rdkit.Chem import MACCSkeys, rdFingerprintGenerator
</pre><p>(remove space)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/348/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (dominiquesydow) [inline teachopencadd/talktorials/T007_compound_activity_machine_learning/talktorial.ipynb:210]: <pre class="ql-syntax" spellcheck="false">Line #29.            fpg = rdFingerprintGenerator.GetMorganGenerator(radius=2,fpSize=n_bits)
</pre><p>as</p><pre class="ql-syntax" spellcheck="false">Line #29.            fpg = rdFingerprintGenerator.GetMorganGenerator(radius=2, fpSize=n_bits)
</pre><p>(add space)</p><p> </p><p>-- same in the call below for radius 3</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/348/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

File: teachopencadd/talktorials/T022_ligand_based_screening_neural_network/talktorial.ipynb [hunk 2]
```diff
@@ -362,7 +362,7 @@
   },
```
Comment (dominiquesydow) [inline teachopencadd/talktorials/T022_ligand_based_screening_neural_network/talktorial.ipynb:362]: <p>As mentioned before, add space in both GetMorganGenerator cases after radius parameter, thank you!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/volkamerlab/teachopencadd/pull/348/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

**Review by dominiquesydow — APPROVED** [review]
Hi @hamzaibrahim21, 
Thanks a lot! Only requested changes are nit-pits really as in adding/removing spaces in order to be in line with our formatting. Once changes, happy to go ahead with the merge if @mbackenkoehler agrees.
I did merge Greg's changes for T004 into this branch here (~let me double-check in a moment if I missed any spaces there~ -> I checked, T004 all good & ready to go from my side).

@mbackenkoehler, I did not find another way to deal with our merge conflicts in T004 than merging changes and resolving conflicts. This is not pretty regarding the commit history but I did not know how to do this better, sorry.

**Comment by mbackenkoehler** [comment]
@hamzaibrahim21 FYI, you can just run `black -l 99 talktorial.ipynb` to automatically format the notebook correctly.

**Pull request merged**